context.go

```diff
@@ -310,17 +310,15 @@ func (c *context) ParamNames() []string {
 
 func (c *context) SetParamNames(names ...string) {
 	c.pnames = names
+	*c.echo.maxParam = len(names)
 }
 
 func (c *context) ParamValues() []string {
 	return c.pvalues[:len(c.pnames)]
 }
 
 func (c *context) SetParamValues(values ...string) {
-	// NOTE: Don't just set c.pvalues = values, because it has to have length c.echo.maxParam at all times
-	for i, val := range values {
-		c.pvalues[i] = val
-	}
+	c.pvalues = values
```

Comment:
Is it safe to set values without checking for the length of pnames here?

Reply:
Param checks the length, so it's OK.
https://github.com/labstack/echo/blob/f93ba1fb9ed3e3d4904151a80a5bae7426ed2fbd/context.go#L298

ParamValues does not check the length, so setting a slice shorter than pnames will panic.
https://github.com/labstack/echo/blob/f93ba1fb9ed3e3d4904151a80a5bae7426ed2fbd/context.go#L317

However, in the documentation it is written to set the same number of values, so it seems to be understood as a specification.
https://echo.labstack.com/guide/testing#setting-path-params
SetParamValues does not return an error, so if you want to fix it, you should check the length with ParamValues.